feat: add metadata and github actions docs for claude code tracing [closes LSDK-161] (#3559)
langchain-ai/docs: src/langsmith/trace-claude-code.mdx

- Each message to Claude Code appears as a trace.
- All turns from the same Claude Code session are grouped using a shared `thread_id` and can be viewed in the **Threads** tab of a project.

## Custom metadata

Set the `CC_LANGSMITH_METADATA` environment variable to a JSON object to attach custom metadata to all traced runs. This is useful for tagging traces with contextual information such as PR URLs, authors, or environment names.

<Tabs>

<Tab title="Settings file (recommended)">

```json
{
  "env": {
    "TRACE_TO_LANGSMITH": "true",
    "CC_LANGSMITH_API_KEY": "<LangSmith API key>",
    "CC_LANGSMITH_PROJECT": "my-project",
    "CC_LANGSMITH_METADATA": "{\"author\":\"jane\",\"environment\":\"development\"}"
  }
}
```

</Tab>

<Tab title="Shell environment variable">

```bash
export CC_LANGSMITH_METADATA='{"author":"jane","environment":"development"}'
```

</Tab>

</Tabs>

> The metadata keys and values will appear on all runs in LangSmith and can be used to filter and search traces.

Reviewer: ```suggestion
The metadata keys and values will appear on all runs in LangSmith, which you can use to filter and search traces.
```
